ql/time/calendars/india.cpp

```diff
@@ -423,39 +424,61 @@ namespace QuantLib {
                 // Diwali - Balipratipada
                 || (d == 14 && m == November)
                 // Gurunank Jayanti
-                || (d == 27 && m == November))
+                || (d == 27 && m == November)
+                // Christmas Day
+                || (d == 25 && m == December))
                 return false;
         }
 
         if (y == 2024) {
-            if ( // Special holiday
+            if ( // Public Holiday
                 (d == 22 && m == January)
-                // Chatrapati Shivaji Jayanti
-                || (d == 19 && m == February)
+                // Republic Day
+                || (d == 26 && m == January)
```

Comment:
Hi,
I'm not sure all of these need to be added as they duplicate the base holiday logic at the top:
<img width="171" alt="image" src="https://github.com/lballabio/QuantLib/assets/43887343/301b186b-84f7-4f4d-b664-566852f3a6ab">

I find it adds complexity here. Without adding them, we are already seeing correct holidays for 2023 and 2024, excluding the Muharrem holiday in July this year, as listed here: https://www.nseindia.com/resources/exchange-communication-holidays. If BBG's INR calendar is different, perhaps we could do with adding another market in the enum within this class?

Reply:
Would that correspond to "Trading holidays" and "Clearing holidays" at https://www.nseindia.com/resources/exchange-communication-holidays?

Reply:
Let me check...

Reply:
That may be the case. I'll gladly help verify when I have access to Bloomberg again after the holidays.  

I still hope that we could simplify the logic, dropping those already covered by the standard annual holidays and some others that fall on weekends in 2024:
<img width="484" alt="image" src="https://github.com/lballabio/QuantLib/assets/43887343/a1d61547-b8a1-445d-8251-d328b432a567">
Would it should suffice to only mention them in the comments for 2024?

There is also a recent announcement for an election holiday in May: https://www.livemint.com/market/stock-market-news/lok-sabha-elections-2024-indian-bourses-likely-to-remain-shut-on-polling-day-in-mumbai-11710637030326.html. This can probably be added as well. 🙂

Reply:
Hi again,
I finally got down to reviewing the calendars. 

I found one historical occasion where the RBI (Mumbai branch) differed from the NSE (clearing holidays). E.g. 2022-02-07 was a holiday in the former, though not the latter. This caused some [turmoil](https://www.timesnownews.com/business-economy/markets/how-maha-decision-to-declare-public-holiday-after-lata-mangeshkars-demise-rattled-foreign-banks-article-90696796), apparently.

Unless there are objections, I'd be happy to incorporate the improvements from this PR in a new PR, with proper attributions of course. Then I can also get the election holiday (2024-05-20) added. Would that be ok with you?

Reply:
Thank you. Unfortunately, I have not had time to look in that. Feel free to do so.